### ISSUE TITLE
Android 12: Need matrix.to/.well-known/assetlinks.json for API-Link URLs

### ISSUE THREAD

**Issue by artisan-digital** [vector-im/element-android#5560]
### Steps to reproduce

This is actually an Android 12 upgrade requirement.  To reproduce, attempt to open an Element URL in a web browser -- nothing happens.  Example URL:
 https://matrix.to/#/!X--------------BDC:matrix.org/$TE--------------------------------------IKU)

![Screenshot_20220316-154311](https://user-images.githubusercontent.com/25208440/158691948-4ea95566-c597-4063-8285-d7d2cfadd6cb.png)

### Outcome

Because "matrix.to" links would not open in Element on Android 12, I had to manually approve URLs in the Android System Apps Electron settings to allow web links to open in Element.  After I approved "matrix.to" in the Android System Apps Element settings, the links open correctly in Element.

According to:

https://developer.android.com/training/app-links/verify-site-associations

this happens because the following URL does not exist:

https://matrix.to/.well-known/assetlinks.json

![Screenshot_20220316-153654_2](https://user-images.githubusercontent.com/25208440/158690802-944aba31-ac4e-4e6c-a948-5342a3404825.png)

### Your phone model

Pixel 3a

### Operating system version

Android 12, build SQ1A.220205.002

### Application version and app store

1.4.2 [40104020] (F-f10c67d3), Matrix SDK: 1.4.2 (f10c67d3), olm: 3.2.10

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by robovoice1** [comment]
Same here.
On the "open it here link" nothing happens (1st pic)

I can add up to 6 links in Os app settings
Please screenshot.
![Screenshot_20220317-144759](https://user-images.githubusercontent.com/92784377/158824232-bafdb86a-ea3c-4647-9add-2e6ab7a503fb.png)

Pixel 6 pro
Android 12 GrapheneOS (latest beta release)
1.4.4 Element app (fdroid)

**Comment by artisan-digital** [comment]
Yes, confirming that the workaround is to tick the six boxes in the System Apps settings.  It's just that most people won't know to do this.

The actual solution is to add the URL file `matrix.to/.well-known/assetlinks.json` to the matrix.to and other servers.

**Comment by notramo** [comment]
This is a small effort - huge impact issue.
